### ISSUE TITLE
Update the client as discussed in PR 507

### ISSUE THREAD

**Issue by shamblett** [shamblett/mqtt_client#510]
Update the client from the suggestions in #507

**Comment by shamblett** [comment]
Specifically address -
```
The examples are slightly misleading here in that they show subscription before listening, I will update these to reverse this to make it clearer if you think it would be better but you are the first to raise this so I guess it must be clear to all the other users.

Looking at the disconnect() method we don't seem to explicitly cancel the updates stream, I think this should be added, I'll do this also.
```

**Comment by shamblett** [comment]
Updated, package re published at version 10.1.0
